test/jdk/java/lang/StringBuffer/IndexOf.java

```diff
@@ -27,162 +27,396 @@
+            + sourceBuffer.indexOf(targetString));
+        failCount++;
+      }
+      if ((sourceBuffer.indexOf(targetString, 5) != index1) ||
+          (index1 != naiveFind(sourceBuffer.toString(), targetString, 0))) {
+        System.err.println("sourceBuffer.indexOf(targetString, 5) fragment '" + targetString + "' ("
+            + targetString.length() + ") String = "
+            + sourceBuffer.toString() + " len Buffer = " + sourceBuffer.toString().length());
+        System.err.println("  naive = " + naiveFind(sourceBuffer.toString(), targetString, 0) + ", IndexOf = "
+            + sourceBuffer.indexOf(targetString, 5));
+        failCount++;
+      }
+      if ((sourceBuffer.indexOf(targetString, 99) == index1) ||
+          (index1 != naiveFind(sourceBuffer.toString(), targetString, 0))) {
+        System.err.println("sourceBuffer.indexOf(targetString, 99) fragment '" + targetString + "' ("
+            + targetString.length() + ") String = "
+            + sourceBuffer.toString() + " len Buffer = " + sourceBuffer.toString().length());
+        System.err.println("  naive = " + naiveFind(sourceBuffer.toString(), targetString, 0) + ", IndexOf = "
+            + sourceBuffer.indexOf(targetString, 99));
+        failCount++;
+      }
+      if ((sourceBuffer.indexOf(emptyString, 99) != 99) ||
+          (99 != naiveFind(sourceBuffer.toString(), emptyString, 99))) {
+        System.err.println("sourceBuffer.indexOf(emptyString, 99) fragment '" + emptyString + "' ("
+            + emptyString.length() + ") String = "
+            + sourceBuffer.toString() + " len Buffer = " + sourceBuffer.toString().length());
+        System.err.println("  naive = " + naiveFind(sourceBuffer.toString(), emptyString, 99) + ", IndexOf = "
+            + sourceBuffer.indexOf(emptyString, 99));
+        failCount++;
+      }
+      if ((allAsBuffer.substring(1, 3).indexOf(allAsBuffer.substring(5, 12)) != -1) ||
+          (-1 != naiveFind(allAsBuffer.substring(1, 3).toString(), allAsBuffer.substring(5, 12), 0))) {
+        System.err.println("allAsBuffer.substring(1, 3).indexOf(allAsBuffer.substring(5, 12)) fragment '"
+            + allAsBuffer.substring(5, 12) + "' ("
+            + allAsBuffer.substring(5, 12).length() + ") String = "
+            + allAsBuffer.substring(1, 3) + " len Buffer = " + allAsBuffer.substring(1, 3).length());
+        System.err.println(
+            "  naive = " + naiveFind(allAsBuffer.substring(1, 3).toString(), allAsBuffer.substring(5, 12), 0)
+                + ", IndexOf = " + allAsBuffer.substring(1, 3).indexOf(allAsBuffer.substring(5, 12)));
+        failCount++;
+      }
+    }
 
-            if (sAnswer != sbAnswer)
-                failCount++;
+    report("Basic Test                   ", failCount);
+  }
+
+  // Note: it is possible although highly improbable that failCount will
+  // be > 0 even if everthing is working ok
```

Comment:
This sounds like either a bug or a testcase bug? Same as line 301, `extremely remote possibility of > 1 match`?

Reply:
This was there from the original author.  I think they were trying to infer that a match could occur in the rare case that the same random string was produced.  They're random after all, and there's no reason the same sequence could be generated.